src/log4stash/Configuration/RequestParameterDictionary.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+
+namespace log4stash.Configuration
+{
+    public class RequestParameterDictionary : IDictionary<string, string>
```

Comment:
In my opinion there is no need to implement `IDictionary` here.
You are just "hiding" a regular dictionary.. I think the only function which is needed here is `AddParameter`.

I saw also that you are using `RequestParameters.ToDictionary` (and I think this is the reason you had to implement  IDictionary), instead you can add another instance function to `RequestParameterDictionary` class and this function will return the desired result.